sogs/routes/users.py

```diff
@@ -98,18 +125,59 @@ def set_mod(sid):
                     app.logger.error("Internal error: unhandled mod/admin room case")
                     raise RuntimeError("Internal error: unhandled mod/admin room case")
 
-        else:  # global mod
-            if visible is None:
-                visible = False
-
-            if (admin, mod) in ((True, None), (None, True)):
-                user.set_moderator(added_by=g.user, admin=admin, visible=visible)
-            elif (admin, mod) == (None, False):
-                user.remove_moderator(removed_by=g.user)
-            elif (admin, mod) == (False, None):
-                user.remove_moderator(removed_by=g.user, remove_admin_only=True)
-            elif (admin, mod) == (False, True):
+    else:  # global mod
+        if visible is None:
+            visible = False
+
+        if (admin, mod) in ((True, None), (None, True)):
+            user.set_moderator(added_by=g.user, admin=admin, visible=visible)
+        elif (admin, mod) == (None, False):
+            user.remove_moderator(removed_by=g.user)
+        elif (admin, mod) == (False, None):
+            user.remove_moderator(removed_by=g.user, remove_admin_only=True)
+        elif (admin, mod) == (False, True):
+            with db.transaction():
                 user.remove_moderator(removed_by=g.user, remove_admin_only=True)
                 user.set_moderator(added_by=g.user, admin=bool(admin), visible=visible)
 
     return jsonify({})
+
+
+@users.post("/user/<SessionID:sid>/ban")
+@auth.user_required
+def ban_user(sid):
+
+    user = User(session_id=sid)
```

Comment:
It needs to get inserted if it doesn't exist, so that I can add the ban row (so that you can preemptively ban someone who has never actually hit the server yet).

Reply:
ok